### ISSUE TITLE
Why is fusion restricted here in dynamic mode?

### ISSUE THREAD

**Issue by Jin-TaoZhang** [pytorch/pytorch#169106]
https://github.com/pytorch/pytorch/blob/3ab08946d5052eaeda11d683d6a58e801a032755/torch/_inductor/ir.py#L3555

I wrote a small demo myself and the numerical accuracy is perfect 
```python
import torch
from torch import nn
from typing import List

#concat in dynamic dim
class MyCatMul(nn.Module):
    def __init__(self, n: int):
        super().__init__()
        self.n = n
        self.W = nn.Parameter(torch.randn(64, 64))

    def forward(self, xs: List[torch.Tensor]):
        assert len(xs) == self.n, f"need {self.n} tensors"
        # last = torch.sigmoid(xs[-1] @ self.W)
        last = torch.sigmoid(xs[-1]) 
        outs = list(xs[:-1]) + [last]
        return torch.cat(outs, dim=0)


n = 15
model = MyCatMul(n).cuda()
x_list = []
for i in range(n):
    a = torch.randint(2, 120, (1,)).item() 
    x_list.append(torch.randn(a, 64, device='cuda'))

from torch.export import export, Dim

dynamic_shapes = [
    {0: Dim(f"b{i}", min=1, max=2048), 1: 64}
    for i in range(n)
]

with torch.no_grad():
    out = model(x_list)
    ep = export(model, (x_list,), dynamic_shapes=[dynamic_shapes])
    torch._inductor.aoti_compile_and_package(
        ep, package_path="./model.pt2", 
            inductor_configs={"max_autotune": True, 
                                "epilogue_fusion": True, 
                                "permute_fusion": True, 
                                "max_autotune_pointwise": True,
                                "max_autotune_gemm":True,
                                "freezing":True,
        }
    )
    aot_model = torch._inductor.aoti_load_package("./model.pt2")
    ################diff##############
    for i in range(100):
        test_input = []
        for ii in range(n):
            a = torch.randint(2, 1024, (1,)).item()
            test_input.append(torch.randn(a, 64, device='cuda'))

        out_raw = model(test_input)
        out_aot = aot_model(test_input)

        diff = torch.abs(out_raw - out_aot)
        max_val, max_idx = diff.max(), diff.argmax()
        coord = torch.unravel_index(max_idx, out_raw.shape)
        val_raw = out_raw.flatten()[max_idx]
        val_aot = out_aot.flatten()[max_idx]
        avg_err = diff.mean().item()
        if max_val > 1e-5:
            print(f"iter {i}: max_err {max_val.item():.8f} @ coord {coord}  "
                f"raw={val_raw.item():.8f}  aot={val_aot.item():.8f}  "
                f"avg_err {avg_err:.8f}")
            raise "error"

    print("pass")


```

cc @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo

**Comment by williamwen42** [comment]
cc @mlazos since you added this code?

**Comment by Jin-TaoZhang** [comment]
> I think this might be able to be removed - combokernels should have fixed this.

Thanks for the explanation — much appreciated!
